### ISSUE TITLE
Error on init path

### ISSUE THREAD

**Issue by sergiorjsd** [elixir-cldr/cldr_collation#3]
Hello!

First of all, thank you for your work here. It's really helpful.

I'm opening this issue because I think I have found a bug while trying to init this library. This bug happens when you have your app compiled and you rename the folder where it is located. Example: I have my `~/projects/myapp` and I rename it to `~/projects/mypersonalapp`.

At this point the library is not working and I think that the problem is [here](https://github.com/elixir-cldr/cldr_collation/blob/master/lib/cldr_collation.ex#L10) and I think that could be solved modifying the `init/0` function as follows:

```
def init do
    so_path = :code.priv_dir(:ex_cldr_collation) ++ '/ucol'
    num_scheds = :erlang.system_info(:schedulers)

    :ok = :erlang.load_nif(so_path, num_scheds)
end
```

instead of using the module attribute `@so_path`.  Do you agree?

I've seen this approach in other libraries like [AppSignal](https://github.com/appsignal/appsignal-elixir/blob/main/lib/appsignal/nif.ex#L13).

I found the problem while trying to deploy my application where we build the application in a `/tmp` folder and after that we move it to the `/app` folder.

**Comment by kipcole9** [comment]
@sergiorjsd Thanks for the report!  I have just pushed commit [9d9227f](https://github.com/elixir-cldr/cldr_collation/commit/9d9227fc7b60185418c3ae3a489b682d94fa4063) that moves the `.so` path from compile time to runtime. Would you check and confirm that this fixes the issue you see?  You'll need to change your `mix.exs` to the following until I publish a new version.

```elixir
def deps do
  [
    ...
    {:ex_cldr_collations, github: "elixir-cldr/cldr_collations"},
    ...
  ]
end
```

The current changelog entry is:

### Bug fixes

* Fix the .so path for the NIF at load time, not compile time. Thanks to @sergiorjsd for the report. Closes #3.

* Fix buidling on ARM-based Mac models

Thanks for the report and kind works - it helps with motivation and is much appreciated.

**Comment by sergiorjsd** [comment]
Yep! Working like a charm now. Thank you so much!

When I tried to test this solution in a fork I couldn't download the library in `mix.exs` from github in a Debian but I don't know why it worked from `Hex PM` :smile: This is now also fixed. I think by the addition of this [change](https://github.com/elixir-cldr/cldr_collation/commit/9d9227fc7b60185418c3ae3a489b682d94fa4063#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947R27).
